ioSPI/fourier.py

```diff
@@ -0,0 +1,237 @@
+import numpy as np
+import pyfftw.interfaces.numpy_fft
+from numba import jit
+
+
+@jit
+def make_neg_pos_2d(arr2d_shape):
+    """Apply checkerboard pattern for 2D FFT.
+
+    Each pixel switches from positive to negative in checker board pattern.
+    Equivalent to fft shifting
+    https://dsp.stackexchange.com/questions/9039/centering-zero-frequency-for-discrete-fourier-transform
+
+    Parameters
+    ----------
+    arr2d_shape : tuple (n_particle,N_fourier_pixels,N_fourier_pixels)
+        Input shape tuple.
+
+    Returns
+    -------
+    neg_pos_2d : numpy.ndarray, shape (n_particle,N_fft_pix,N_fft_pix)
+        Checkerboard array.
+    """
+    # assert arr2d.ndim == 3  # extra axis
+    n_particle, R1, R2 = arr2d_shape
```

Comment:
Uppercase letters denote "constants" in Python. thus, there should be `r1` and r2`.

Additionally, what do r1 and r2 represent? any chance that we could give them another name to help the reader go through your code?